kernel/init.c

```diff
@@ -389,15 +389,20 @@ u32_t z_early_boot_rand32_get(void)
 		goto sys_rand32_fallback;
 	}
 
-	rc = entropy_get_entropy(entropy, (u8_t *)&retval, sizeof(retval));
-	if (rc == 0) {
-		return retval;
+	/* Try to see if driver provides an ISR-specific API */
+	rc = entropy_get_entropy_isr(entropy, (u8_t *)&retval,
+				     sizeof(retval), ENTROPY_BUSYWAIT);
+	if (rc < 0) {
+		/* Driver does not provide an ISR-specific API, assume it can
```

Comment:
Shouldn't this only happen if `rc == -ENOSUP`?  The implementation might support the ISR version, and return other errors (e.g. not enough entropy at that time and no ENTROPY_BUSYWAIT passed)...

Reply:
Yes, you are probably right. Let me take another look